system/classes/KO7/REST/Format/XML.php

```diff
@@ -0,0 +1,113 @@
+<?php
+/**
+ * Class for formatting REST-Response bodies as XML
+ *
+ * @package        KO7\REST
+ *
+ * @copyright  (c) since 2016 Koseven Team
+ * @license        https://koseven.ga/LICENSE
+ */
+class KO7_REST_Format_XML extends REST_Format {
+
+    /**
+     * Format function
+     *
+     * @throws REST_Exception
+     *
+     * @return string
+     */
+    public function format() : string
+    {
+        // Check if php-xml is loaded
+        if ( ! extension_loaded('xml'))
+        {
+            throw new REST_Exception('PHP XML Module not loaded.');
+        }
+
+        // Create new XML Element
+        $xml = $this->array_to_xml($this->_body);
+
+        // Check if xml is valid
+        if ( ! $xml)
+        {
+            throw new REST_Exception($this->evaluate_error());
+        }
+
+        return $xml;
+    }
+
+    /**
+     * Convert Array to an xml element
+     *
+     * @param array $array        Array to convert
+     * @param mixed $rootElement  Root element of entry
+     * @param null $xml           Current XML stack
+     *
+     * @return mixed
+     */
+    private function array_to_xml(array $array, $rootElement = NULL, $xml = NULL)
+    {
+        // If there is no Root Element then insert root
+        if ($xml === NULL)
+        {
+            $xml = new SimpleXMLElement($rootElement ?? '<?xml version="1.0" encoding="'.KO7::$charset.'"?><root/>', LIBXML_COMPACT);
```

Comment:
1. u cant catch error as u want because remove `LIBXML_NOWARNING | LIBXML_NOERROR`
2. `LIBXML_BIGLINES` too important
3. dont set methods as `private` - hard to extending this classes

Reply:
1. why?
2. Added
3. Moved to protected

Reply:
Sure..got the first one...no worries....fixing it with libxml_use_internal_errors(true)